File: backend/models/user.model.js
module.exports = (sequelize, Sequelize) => {
    const User = sequelize.define('users', {
        email: {
            type: Sequelize.STRING
        },
        password: {
            type: Sequelize.STRING
        },
        approved: {
            type: Sequelize.BOOLEAN
        },
        role: {
            type: Sequelize.STRING
        }
    });
    return User;
}

File: frontend/lib/domain/auth/user_model.dart
class User {
  final int? id;
  final String email;
  final String? password;
  final String? role;
  final bool? approved;

  User({
    this.id,
    required this.email,
    this.password,
    required this.role,
    this.approved,
  });

  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      id: json['id'],
      email: json['email'],
      password: json['password'],
      role: json['role'],
      approved: json['approved'],
    );
  }
}
